e2e/test/netstandard20/ProvisioningE2ETests.cs

```diff
@@ -493,51 +641,68 @@ private Client.TransportType GetDeviceTransportType(string provisioningTransport
-
-                    string base64Ek = Convert.ToBase64String(tpmSim.GetEndorsementKey());
-                    string registrationId = Configuration.Provisioning.TpmDeviceRegistrationId;
-
-
-                    var provisioningService = ProvisioningServiceClient.CreateFromConnectionString(Configuration.Provisioning.ConnectionString);
-
-                    _log.WriteLine($"Getting enrollment: RegistrationID = {registrationId}");
-                    IndividualEnrollment enrollment = await provisioningService.GetIndividualEnrollmentAsync(registrationId).ConfigureAwait(false);
-                    var attestation = new TpmAttestation(base64Ek);
-                    enrollment.Attestation = attestation;
-                    _log.WriteLine($"Updating enrollment: RegistrationID = {registrationId} EK = '{base64Ek}'");
-                    await provisioningService.CreateOrUpdateIndividualEnrollmentAsync(enrollment).ConfigureAwait(false);
+            var provisioningServiceClient = ProvisioningServiceClient.CreateFromConnectionString(Configuration.Provisioning.ConnectionString);
 
+            switch (attestationType)
+            {
+                case AttestationType.Tpm:
+                    IndividualEnrollment tpmIndividualEnrollment = await CreateIndividualEnrollment(provisioningServiceClient, AttestationType.Tpm).ConfigureAwait(false);
+                    var tpmSim = new SecurityProviderTpmSimulator(tpmIndividualEnrollment.RegistrationId);
                     return tpmSim;
 
-                case nameof(SecurityProviderX509Certificate):
+                case AttestationType.x509:
 
                     X509Certificate2 certificate = null;
                     X509Certificate2Collection collection = null;
-                    switch (x509Type)
+                    switch (enrollmentType)
                     {
-                        case X509EnrollmentType.Individual:
+                        case EnrollmentType.Individual:
                             certificate = Configuration.Provisioning.GetIndividualEnrollmentCertificate();
                             break;
-                        case X509EnrollmentType.Group:
+                        case EnrollmentType.Group:
                             certificate = Configuration.Provisioning.GetGroupEnrollmentCertificate();
                             collection = Configuration.Provisioning.GetGroupEnrollmentChain();
                             break;
                         default:
-                            throw new NotSupportedException($"Unknown X509 type: '{x509Type}'");
+                            throw new NotSupportedException($"Unknown X509 type: '{enrollmentType}'");
                     }
 
                     return new SecurityProviderX509Certificate(certificate, collection);
+
+                case AttestationType.SymmetricKey:
+                    if (enrollmentType == EnrollmentType.Group)
```

Comment:
nit: use switch case for enrollmentType (consistency with x509 case above).

Reply:
sure